txnsync/peer.go

```diff
@@ -215,6 +211,11 @@ func (p *Peer) selectPendingTransactions(pendingTransactions []transactions.Sign
 			}
 		}
 
+		if p.recentSentTransactions.contained(txID) {
+			// we already sent that transaction. no need to send again.
+			continue
```

Comment:
When can this happen?

Reply:
The other node send us periodically a new bloom filter. When this happen, we fetch a refreshed list of transactions from the transaction pool, and filter these using this method. The refreshed list might contain some of the transactions that were previously sent to the peer. We can't use the bloom filter for that, since the bloom filter isn't being updated frequently enough.

Reply:
Is this assuming that a transaction only needs to be evaluated once during the selection process?

What about a multiple spend case. If relay A processes txA which drains an account and relay B processes txB which also drains the account. Is this optimization still valid to skip?

Reply:
multiple spend ( or transaction validity in general ) is not something the transaction sync would try/need to address. The goal fo the transaction sync is to make sure the transaction gets to all intended parties( ideally only once). The validation is done on the transaction pool side.